tests/utils/index.js

```diff
@@ -526,15 +526,25 @@ const waitForSettingsUpdateLogs = (type) => {
  *                                       api logs, if value equals 'sentinel', will watch sentinel logs instead.
  * @return {Promise}        completion promise
  */
-const updateSettings = async (updates, ignoreReload) => {
-  const watcher = ignoreReload &&
-    Object.keys(updates).length &&
-    await waitForSettingsUpdateLogs(ignoreReload);
+
+const updateSettings = async (updates, options = {}) => {
+  const {ignoreReload = false, sync = false, refresh = false, revert = false} = options;
+  if (revert) {
+    await revertSettings(true);
+  }
+  const watcher = ignoreReload && Object.keys(updates).length && await waitForSettingsUpdateLogs(ignoreReload);
   await updateCustomSettings(updates);
   if (!ignoreReload) {
-    return await commonElements.closeReloadModal(true);
+    await commonElements.closeReloadModal(true);
+  } else if (watcher) {
+    await watcher.promise;
+  }
+  if (sync) {
+    await commonElements.sync(true);
+  }
+  if (refresh) {
+    await browser.refresh();
```

Comment:
Cool! This cleans up code quite nicely. One comment is that when we choose not to ignoreReload, it means that we have an online user that will never have the sync button, and the "reload" modal will also refresh the page. So we can just return after `await commonElements.closeReloadModal(true);` to make the function simpler.
I'm not sure if we will ever have `sync: false` and `refresh: true`, we could consider connecting these somehow.

Reply:
Until now, we have used these options in three ways:
`sync:false` and` refresh:false` (default values)
`sync:true` and `refresh:true`
`sync:true `and `refresh: false` (only once in an e2e default tasks test)

I can implicitly set the sync option to true if a refresh is true, but I think it is better to leave the flexibility. What do you think?

Reply:
If flexibility is non-sensical it can also create confusion, because then we would never have function parameters, we would just pass unlimited args to every function. In the end, this isn't so important in this case.